### DISCUSSION TITLE
Fields specified as required even though they are set as Optional

### DISCUSSION THREAD

**Discussion by josushiman** [collerek/ormar#249] [Q&A] [answered]
Hello,

Following on from the fix implemented in https://github.com/collerek/ormar/discussions/220#discussion-3394029, I am trying to create a Country with no borders referenced, however I seem to be getting Validation issues when attempting to create a Country. This is even though I have the borders field set to Optional, and with a default value of None.

I am not sure if I am doing this correctly, but any assistance would be greatly appreciated!

Thanks in advance

Model:
```
# models.py
class Country(ormar.Model):
    class Meta(BaseMeta):
        tablename = "countries"
        constraints = [ormar.UniqueColumns("name"), ormar.UniqueColumns("iso2")]

    id: int = ormar.Integer(primary_key=True)
    name: str = ormar.String(max_length=128)
    iso2: str = ormar.String(max_length=3)
    iso3: str = ormar.String(max_length=4)
    population: int = ormar.Integer(maximum=10000000000)
    demonym: str = ormar.String(max_length=128)
    native_name: str = ormar.String(max_length=128)
    capital: Optional[CityRef] = ormar.ForeignKey(CityRef, related_name="capital_city", nullable=True)
    borders: Optional[List[CountryRef]] = ormar.ManyToMany(CountryRef, default=None, skip_reverse=True)
```

Country POST handling with crud helper function:
```
# countries.py
@router.post("/", response_model=Country, status_code=status.HTTP_201_CREATED)
async def create_country(country: Country):
    return await db.create_update_entity(country)

async def create_update_entity(model: Model) -> Model:
       try:
            EntityModel = cls.get_entity_model(model)
            return await EntityModel.objects.update_or_create(**entity_kwargs)
        except Exception as e:
            print("Issue creating/updating entity")
            print(e)
            raise HTTPException(status_code=422, detail=f"Unable to create/update {model.__name__} entity.")
```

POST example:
```
{
    "name": "Thailand",
    "iso2": "TH",
    "iso3": "THA",
    "population": 23123123,
    "demonym": "Thai",
    "native_name": "Thailand"
}
```

Exception/Error raised:
```
Issue creating/updating entity
6 validation errors for Country
borders -> name
  field required (type=value_error.missing)
borders -> iso2
  field required (type=value_error.missing)
borders -> iso3
  field required (type=value_error.missing)
borders -> population
  field required (type=value_error.missing)
borders -> demonym
  field required (type=value_error.missing)
borders -> native_name
  field required (type=value_error.missing)
```

**Comment by collerek** [comment]
OK, so few points:
1. ormar sets the pydantic fields not the other way around -> therefore all type hints are optional (for type checking and auto completion in ide) and changing the type hint does not affect ormar. To set a field as optional you need to set it to `nullable=True` in ormar field declaration, but for relation fields that is already the default (so you don't have to).
2. you cannot set default on relation fields, if you would set it to anything else than None it would raise an exception. Since default values are populated in synchronize part of the code (in example in init), and ormar models needs to be saved in database for proper relation that is not supported.
3.  if country in your endpoint is an ormar model you do not need to process it to entity, simply something like this is enough
```python
@router.post("/", response_model=Country, status_code=status.HTTP_201_CREATED)
async def create_country(country: Country): # if this is ormar
    return await country.upsert() # it's already initialized as ormar model
```

But saying all this there seems to be a bug with pydantic field definition, will check that. Thanks for reporting!

**Reply by josushiman** [reply]
thank you for all the tips and the incredibly quick fix! extremely appreciated

**Comment by collerek** [comment, marked as answer]
Fixed in 0.10.12 (via #251)
